packages/suite/src/actions/settings/deviceSettingsActions.ts

```diff
@@ -175,3 +185,30 @@ export const resetDevice =
 
         return result;
     };
+
+export const changeLanguage = createThunk(
+    `${firmwareActionsPrefix}/update-firmware-language`,
+    async (
+        params: Omit<Parameters<typeof TrezorConnect.changeLanguage>[0], 'baseUrl'>,
+        { dispatch, getState },
+    ) => {
+        const device = selectDevice(getState());
+        const useDevkit = selectUseDevkit(getState());
+        const desktopBinDir = selectDesktopBinDir(getState().desktop);
+
+        if (!device) return;
+
+        // FW binaries are stored in "*/static/connect/data/firmware/*/*.bin". see "connect-common" package
+        const baseUrl = `${isDesktop() ? desktopBinDir : resolveStaticPath('connect/data')}${
+            useDevkit ? '/devkit' : ''
+        }`;
```

Comment:
This snippet and some selectors are copies from `firmwareInstallThunk` - would be nice to reuse it.

Reply:
Fixed